index.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+const cp = require('child_process');
+
+/**
+ * An error to be thrown when invokation of the cumulus-message-adapter fails
+ */
+class CumulusMessageAdapterExecutionError extends Error {
+  // eslint-disable-next-line require-jsdoc
+  constructor(message, fileName, lineNumber) {
+    super(message, fileName, lineNumber);
+    this.name = 'CumulusMessageAdapterExecutionError';
+  }
+}
+
+/**
+ * Invoke the cumulus-message-adapter
+ *
+ * @param {string} command - the action to be performed by the message-adapter
+ * @param {Object} input - the input to be sent to the message-adapter
+ * @returns {Promise.<Object>} - the output of the message-adapter
+ */
+function callCumulusMessageAdapter(command, input) {
+  return new Promise((resolve, reject) => {
+    const sled = cp.spawn('python', ['./cumulus-sled.zip', command]);
+
+    // Collect STDOUT
+    let sledStdout = '';
+    sled.stdout.on('data', (chunk) => {
+      sledStdout += chunk;
+    });
+
+    // Collect STDERR
+    let sledStderr = '';
+    sled.stderr.on('data', (chunk) => {
+      sledStderr += chunk;
+    });
+
+    sled.on('close', (code) => {
+      if (code === 0) resolve(JSON.parse(sledStdout));
+      else reject(new CumulusMessageAdapterExecutionError(sledStderr));
+    });
+
+    // If STDIN is closed already, something went wrong.  Don't handle it
+    // here and expect a non-zero code when the close event fires.
+    // If this line is not present, calling sled.stdin.end with a chunk will
+    // result in an unhandled "Error: write EPIPE".
+    sled.stdin.on('error', () => {});
+
+    sled.stdin.end(JSON.stringify(input));
+  });
+}
+
+/**
+ * If a Cumulus Remote Message is passed, fetch it and return a full Cumulus Message
+ *
+ * @param {Object} cumulusMessage - either a full Cumulus Message or a Cumulus Remote Message
+ * @returns {Promise.<Object>} - a full Cumulus Message
+ */
+function loadRemoteEvent(cumulusMessage) {
+  return callCumulusMessageAdapter('loadRemoteEvent', { event: cumulusMessage });
+}
+
+/**
+ * Query AWS to create a task-specific event
+ *
+ * @param {Object} cumulusMessage - a full Cumulus Message
+ * @param {Object} context - an AWS Lambda context
+ * @returns {Promise.<Object>} - an Object containing the keys input, config and messageConfig
+ */
+function loadNestedEvent(cumulusMessage, context) {
+  return callCumulusMessageAdapter('loadNestedEvent', {
+    event: cumulusMessage,
+    context
+  });
+}
+
+/**
+ * Query AWS to create a task-specific event
```

Comment:
Should change the comment to differentiate from the function above.

Reply:
Fixed.